core/src/impl/Kokkos_HostSharedPtr.hpp

```diff
@@ -56,12 +56,14 @@ namespace Impl {
 template <typename T>
 class HostSharedPtr {
  public:
+  HostSharedPtr(T* value_ptr = nullptr)
+      : m_value_ptr(value_ptr), m_deleter(nullptr), m_counter(nullptr) {}
```

Comment:
That makes the `HostSharedPtr<T> p(new T())` leak...

Reply:
Yes, this interface assumes that the one-argument constructor is non-owning.